src/python/pants/option/global_options.py

```diff
@@ -216,6 +216,11 @@ def register_bootstrap_options(cls, register):
              help='The path to the watchman UNIX socket. This can be overridden if the default '
                   'absolute path length exceeds the maximum allowed by the OS.')
 
+    # This option changes the parser behavior in a fundamental way (which currently invalidates
+    # all caches), and needs to be parsed out early, so we make it a bootstrap option.
+    register('--build-file-imports', choices=['allow', 'warn', 'error'], default='allow',
```

Comment:
Let's default to warning for this so we can start recommending evolution away from imports now.

Reply:
Done